h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -838,4 +838,66 @@ public void map(Chunk c, NewChunk nc) {
     }
   }
 
+  /**
+   * shuffle values of a Vec {@link Vec}.
+   * 
+   * The 'new' values of the vector will be the already exisitng one's however randomly shuffeled
+   * 
+   * 
+   */
+  
+  public static class shuffleVec extends MRTask<shuffleVec>{
+    public Vec _vec;
+        
+    public shuffleVec(Vec vec) {this._vec = vec;}
+        
+        @Override
+        public void map(Chunk ch/*, NewChunk nc*/) {
+//          nc.alloc_doubles(ch._len);
+                
+            for (int i = 0; i < ch._len; ++i) {
+                int randIndex = i + (int) (Math.random() * (ch._len - i));
+                double vecVal = ch.atd(i);
+                double nextvecVal = ch.atd(randIndex);
+                _vec.set(i, nextvecVal);      // this
+                _vec.set(randIndex, vecVal);  // and this lines must be expensive! *For now leaving it like this
+            }
+        }
+            
+        @Override 
+        public void reduce(shuffleVec mrt) {
+            if( _vec != mrt._vec ) _vec = mrt._vec;
+        }
+
+    }
```

Comment:
So depends on 1., 2. and 3. you need to rewrite your MRTask to shuffle only values in the input chunk. You can shuffle the original chunk, or you have to use `NewChunk nc` to save the output, but then you should deal with rows that can be unchanged. In the reduce phase all chunks are connected back to the final frame.  If you dont want random shuffle some values across chunks, you cannot implement `public void reduce(shuffleVec mrt)` method.

Reply:
A simple example to see what is happening insight MRTask: 

You have input vector [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12] and three nodes. The chunk size is 4. 

- node 1 gets chunk with [1, 2, 3, 4] data
- node 2 gets chunk with [5, 6, 7, 8] data
- node 3 gets chunk with [9, 10, 11, 12] data.

Every node randomly shuffles its data. So, for example:

- node 1 has now shuffled chunk with  [3, 4, 1, 2] data
- node 2 has now shuffled chunk with [7, 5, 8, 6] data
- node 3has now shuffled chunk with [12, 11, 9, 10] data.

In the reduce phase,  all chunks are connected to one result frame: [3, 4, 1, 2, 7, 5, 8, 6, 12, 11, 9, 10], or a few items can be shuffled across all chunks. 

This is example of just a naive implementation to demonstrate how it works. I hope there are a plenty of algorithms how to distributed shuffle a vector.